opencog/atoms/core/Variables.cc

```diff
@@ -709,15 +709,15 @@ void Variables::get_vartype(const Handle& htypelink)
 		Type vt = TypeNodeCast(vartype)->get_kind();
 		TypeSet ts;
 		TypeSet::iterator it = ts.begin();
-		nameserver().getChildren(vt, std::inserter(ts, it));
+		nameserver().getChildrenRecursive(vt, std::inserter(ts, it));
 		_simple_typemap.insert({varname, ts});
```

Comment:
OK, the above suggests that there should be a method
```
TypeSet NameServer::getChildrenRecursive(Type vt) {...}
```
and then you could just say
```
_simple_typemap.insert({varname, namserver(),getChildrenRecursive(vt)});
```
instead of all that ugly iterator.begin stuff, and the repeated cut-n-paste of this 4-line block of code...

Anyway, such a change is not for this pull req, but for some other one.